internal/probe.go

```diff
@@ -20,6 +20,8 @@ type Probe struct {
 	Logger *slog.Logger
 	// Channel to send back partial results.
 	ReportCh chan *Report
+	// User defined adresses to check
+	Addrs []string
```

Comment:
These could be URLs (HTTP), host/port strings (TCP) or domains (DNS). So I would rename `addrs` (and related) everywhere to `input`.